### PR TITLE
Allow custom copy methods

### PR DESCRIPTION
**Pull request by monikasulik**

### CODE REVIEW

File: djangocms_versioning/models.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from django.apps import apps
```
Comment (chaosk) [inline djangocms_versioning/models.py]: Add unique_together constraint on (content_type, object_id)

File: djangocms_versioning/datastructures.py [hunk 2]
```diff
@@ -6,10 +6,11 @@
```
Comment (chaosk) [inline djangocms_versioning/datastructures.py]: Update docs to reflect new changes.

Comment (monikasulik) [reply djangocms_versioning/datastructures.py]: Done

File: djangocms_versioning/models.py (hunk 1 above)
Comment (monikasulik) [reply djangocms_versioning/models.py]: Done

**Comment by czpython** [comment]
This is great! Perfect execution! 💯 🥇 💯

File: tests/test_models.py [hunk 3]
```diff
@@ -44,3 +73,21 @@ def test_content_object_gets_duplicated(self):
             original_version.content.poll,
             new_version.content.poll,
         )
+
+    def test_the_copy_method_is_configurable(self):
+        """When copying, the new version object should have a new
```
Comment (damilare) [inline tests/test_models.py]: docs in tests, suger!

**Comment by damilare** [comment]
All Hail Lady Monika, you have done incredible well!!! Great Stuff!! :rainbow: :unicorn: :dragon: :dragon_face: :performing_arts:

File: djangocms_versioning/datastructures.py [hunk 4]
```diff
@@ -6,10 +6,11 @@
 
 class VersionableItem:
 
-    def __init__(self, content_model, grouper_field_name):
+    def __init__(self, content_model, grouper_field_name, copy_function):
```
[the remark this answers is not in the document]
Comment (monikasulik) [reply djangocms_versioning/datastructures.py]: We actually had a discussion about this with Krzysztof. The reason is that this way the developer who puts versioning into their project is more conscious of there being such a thing as a copy function. I'm anticipating that they will need to change it perhaps even 80% of the time or else have strange things happen, so I wanted to make this as visible as possible.

**Pull request merged**